clang/lib/Headers/hlsl/hlsl_intrinsics.h

```diff
@@ -80,6 +80,34 @@ void asuint(double3, out uint3, out uint3);
 _HLSL_BUILTIN_ALIAS(__builtin_hlsl_elementwise_splitdouble)
 void asuint(double4, out uint4, out uint4);
 
+//===----------------------------------------------------------------------===//
+// asint16 builtins
+//===----------------------------------------------------------------------===//
+
+/// \fn int16_t asint16(T X)
+/// \brief Interprets the bit pattern of \a X as an 16-bit integer.
+/// \param X The input value.
+#ifdef __HLSL_ENABLE_16_BIT
+
+template <typename T, int N>
+constexpr __detail::enable_if_t<__detail::is_same<int16_t, T>::value ||
```

Comment:
You need to add the [Shader Model 6.2](https://github.com/microsoft/DirectXShaderCompiler/wiki/Shader-Model-6.2) availability macro to these two functions.